lib/pages/home/lib/draft_editing_controller.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:Okuna/services/draft.dart';
+import 'package:flutter/material.dart';
+
+abstract class DraftTextEditingController extends TextEditingController {
+  final DraftService _draftService;
+
+  DraftTextEditingController(this._draftService) {
+    addListener(_textChanged);
+  }
+
+  factory DraftTextEditingController.comment(int postId,
+      {int commentId, String text, @required DraftService draftService}) {
+    return _CommentDraftEditingController(
+        postId, commentId, text, draftService);
+  }
+
+  factory DraftTextEditingController.post(
+      {int communityId, String text, @required DraftService draftService}) {
+    return _PostDraftEditingController(communityId, text, draftService);
+  }
+
+  void _textChanged();
+
+  void clearDraft();
+}
+
+class _CommentDraftEditingController extends DraftTextEditingController {
+  final int postId;
+  final int commentId;
+
+  _CommentDraftEditingController(
+      this.postId, this.commentId, String text, DraftService draftService)
+      : super(draftService) {
+    if (text == null) {
+      this.text = _draftService.getCommentDraft(postId, commentId);
```

Comment:
This feature is cool! Thanks @Komposten 

Some suggestions, if you are using abstract factories/classes, it might be better to do something like this   although its overkill for this feature. 
But abstract classes, should never be instantiated, only extended as a general rule. 

``` 
abstract class DraftTextEditingController extends TextEditingController {    
	 
	 DraftTextEditingController(this._draftService) {
    	addListener(_textChanged);
	}
	
	void _textChanged() ;   
	...
	...
}

```

```
class _CommentDraftEditingController extends DraftTextEditingController {
    final int postId;
    final int commentId;

	  _CommentDraftEditingController(
      this.postId, this.commentId, String text, DraftService draftService)
      : super(draftService) {
	    if (text == null) {
	      this.text = _draftService.getCommentDraft(postId, commentId);
	    }
    }
    ...
    ...
    ...
}

```

Then have a factory which creates your objects for you

```

abstract class DraftControllerFactory() {
    void make(String type, Map config) {}
}

class DraftTypeControllerFactory extends DraftControllerFactory() {
	void make(String type, Map config) {
	    if (type == 'POST') {
	     	return _PostDraftEditingController(config.communityId, config.text, config.draftService);
	    } else if (type == 'COMMENT') {
	    	return _CommentDraftEditingController(config.postId, config.commentId, config.text, config.draftService);
	    }
	}
}


```

In your widgets, you can use this factory

```
_draftTypeControllerFactory = DraftTypeControllerFactory(); 
_textController = _draftTypeControllerFactory.make(type: 'POST', config: {
	'communityId': _communityId,
	'text': _text,
	'draftService': _draftService
});

```

Reply:
So instead of using `factory` methods (or whatever they are called) I should create a factory class when I'm working with an abstract class? Will make a note of that.

For the factory class it feels better, to me, to have two separate "make" methods (like my current factory methods) rather than a very abstract config parameter that would require additional documentation to explain.

Reply:
I meant in general a factory should basically cook up objects of one kind or in this case multiple kinds.(although purists might argue one factory should only give out one kind of object but we can be practical here) I used it since your PR had factory methods so I assumed you wanted to use them ( it also fits the use case)

Abstract class and factories are not tied. You can have just one too. But you were instantiating your abstract class, which is why I refactored things a bit not to do so.  

And yes, I guess you can have two separate make methods too to be explicit 👍

Reply:
Noted for the future 🙂